src/pybind/mgr/cephadm/upgrade.py

```diff
@@ -25,6 +25,9 @@ def __init__(self,
                  target_version: Optional[str] = None,
                  error: Optional[str] = None,
                  paused: Optional[bool] = None,
+                 services_complete: Optional[List[str]] = None,
```

Comment:
`UpgradeState` is special as it needs to cope with up and downgrades. I'd be somewhat careful in adding new fields to `UpgradeState`. IMO, we should only put things that we need in order to upgrade the cluster.

Can we compute `services_complete` at runtime? If yes, we shouldn't add this here.